syntax = "proto3";

package proto_messages;

message Ball
{
  uint32 position_x = 1;
  uint32 position_y = 2;
}

message GameState
{
  string state         = 1;
  uint32 clients_ready = 2;
}

message Paddle
{
  uint32 position = 1;
}

message Score
{
  uint32 left  = 1;
  uint32 right = 2;
}

message PowerUp {
  bool active = 1;
  int32 x = 2;
  int32 y = 3;
  string target = 4; // "left" oder "right"
}

message PongGameData {
  Ball ball = 1;
  Paddle paddle_left = 2;
  Paddle paddle_right = 3;
  Score score = 4;
  GameState game_state = 5;

  PowerUp power_up_left = 6;
  PowerUp power_up_right = 7;
}
